src/platform_linux/laio.c

```diff
@@ -67,40 +70,140 @@ laio_cleanup(io_handle *ioh, uint64 count);
 static void
 laio_cleanup_all(io_handle *ioh);
 
+static void
+laio_register_thread(io_handle *ioh);
+
+static void
+laio_deregister_thread(io_handle *ioh);
+
 static io_async_req *
 laio_get_kth_req(laio_handle *io, uint64 k);
 
 /*
  * Define an implementation of the abstract IO Ops interface methods.
  */
 static io_ops laio_ops = {
-   .read          = laio_read,
-   .write         = laio_write,
-   .get_iovec     = laio_get_iovec,
-   .get_async_req = laio_get_async_req,
-   .get_metadata  = laio_get_metadata,
-   .read_async    = laio_read_async,
-   .write_async   = laio_write_async,
-   .cleanup       = laio_cleanup,
-   .cleanup_all   = laio_cleanup_all,
-   .get_context   = laio_get_context,
+   .read              = laio_read,
+   .write             = laio_write,
+   .get_iovec         = laio_get_iovec,
+   .get_async_req     = laio_get_async_req,
+   .get_metadata      = laio_get_metadata,
+   .read_async        = laio_read_async,
+   .write_async       = laio_write_async,
+   .cleanup           = laio_cleanup,
+   .cleanup_all       = laio_cleanup_all,
+   .register_thread   = laio_register_thread,
+   .deregister_thread = laio_deregister_thread,
+   .get_context       = laio_get_context,
+   .get_io_async_req  = laio_get_io_async_req,
 };
 
+/*
+ * Given an AIO handle, set up a thread-specific IO context opaque handle.
+ */
+static platform_status
+io_context_setup(laio_handle *io)
```

Comment:
New thread-specific setup routine. Review carefully.

Reply:
@rtjohnso and I walked through the logic changes during our review discussion and agreed to the nature of these changes.